NitroxPatcher/Patches/Dynamic/DevConsole_Update_Patch.cs

```diff
@@ -1,19 +1,56 @@
-﻿using HarmonyLib;
+﻿using System.Collections.Generic;
+using System.Reflection;
+using HarmonyLib;
 using NitroxClient.GameLogic;
 using NitroxModel.Helper;
+using NitroxPatcher.PatternMatching;
+using UnityEngine;
+using static System.Reflection.Emit.OpCodes;
 
 namespace NitroxPatcher.Patches.Dynamic
 {
-    class DevConsole_Update_Patch : NitroxPatch, IDynamicPatch
+    internal class DevConsole_Update_Patch : NitroxPatch, IDynamicPatch
     {
+        /// <summary>
+        ///     Pattern to keep DevConsole disabled when enter is pressed.
+        /// </summary>
+        private static readonly InstructionsPattern devConsoleSetStateTruePattern = new()
+        {
+            InstructionPattern.Call(nameof(Input), nameof(Input.GetKeyDown)),
+            Brfalse,
+            Ldarg_0,
+            Ldfld,
+            Brtrue,
+            Ldarg_0,
+            { Ldc_I4_1, "TrueArgument" },
+            InstructionPattern.Call(nameof(DevConsole), nameof(DevConsole.SetState))
+        };
+
+        public static readonly MethodInfo TargetMethod = Reflect.Method((DevConsole t) => t.Update());
+
+        /// <summary>
+        ///     Prevent console from opening if server wants it disabled.
+        /// </summary>
         public static void Prefix()
         {
             PlatformUtils.SetDevToolsEnabled(!NitroxConsole.DisableConsole);
         }
 
+        public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnumerable<CodeInstruction> instructions) => instructions
```

Comment:
Since we are adding a transpiler we should also add an test for it

Reply:
The way transpiler tests are done is dumb. They don't prevent us from breaking the code and neither do the seem to help us fixing it for new game code. The way this pattern system works will error out if the code changes significantly (but at runtime, not CI/build time). 

I'll add it just for posterity but we should evaluate if these tests provide value.